sources/code/main/modules/extensions.ts

```diff
@@ -1,12 +1,19 @@
+import { join, dirname } from "path";
 import { commonCatches } from "./error";
 
 const safeStoragePromise = (import("electron/main"))
   .then(main => main.safeStorage);
 
-async function fetchOrRead(file:string, signal?:AbortSignal) {
+async function fetchOrRead(file:string, prevImport = "", signal?:AbortSignal) {
   const readFile = import("fs/promises").then(fs => fs.readFile);
 
-  const url = new URL(file);
+  // Try to parse the import string as a complete URL. If it fails, prepend the previous path and try again
+  let url;
+  try {
+    url = new URL(file);
+  } catch {
+    url = new URL(join(prevImport, file));
+  }
```

Comment:
Why not eliminate the `try...catch` entirely and just make a use of `resolve` (the code would look a bit more concise IMO and I guess there could be less `URL` calls in case of relative paths, performance wise I don't think there would be a lot of differences)?

Also I guess we could use `resolve` before this function is called. Eliminating `prevImport` might also be possible this way, I think `importCalls` could be a replacement since it basically functions as a stack there (i.e. the last element should inform you about the last file/path that was loaded during a single recursion, as a whole an array can be perceived as an image a single recursion branch). So I think there's a lot of redundancy with the changes you suggest right now.

Reply:
The reason of using two `new URL()`s instead of `url.resolve` is 1. I wanted to eliminate any chance of this being a breaking change by only changing behavior if the URL would be invalid, and 2. `url.resolve` cuts off the last directory when the `from` argument does not end in a slash.

I also didn't use `importCalls` because there is a bug the current implementation that causes any retry to fail.

But since you want it done this way, I changed it to use `url.resolve` anyway and fixed the `importCalls` issue here.

Reply:
> But since you want it done this way, I changed it to use url.resolve anyway and fixed the importCalls issue here.

I've actually considered also using `path.resolve`, but in the reality the use of `try...catch` wasn't that much of concern to me. And I just wanted from you mostly to explain your approach there, so whenever you modified it or not I didn't care as long as you could explain why you find your own approach beneficial. Sorry if that wasn't clean enough.

But the behaviour of `url.resolve` actually made me concerned if it isn't actually something we preffer to have, and if it could eliminate the `dirname` use - but I need to take a while, clone this PR and analyse the code, I have to admit I rather briefly looked at it. So I might be just saying things.

Reply:
I didn't realize that you wanted to use `url.resolve` to avoid `dirname`. I pushed a commit that should accomplish what you want now.